### PR TITLE
Initially added some validation tests for WebNN API operations

### PR DESCRIPTION
**Pull request by BruceDai**
@fdwr @huningxin @honry PTAL firstly. I'd appreciate any feedbacks, thanks.

### CODE REVIEW

File: webnn/resources/utils_validation.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        }
+      } else {
+        // resample2d
+        for (let dataType of operandDataTypeArray) {
+          const input = builder.input('input', {dataType, dimensions: dimensionsArray[inputRank]});
+          for (let outsideAxis of outsideAxisArray) {
+            assert_throws_js(TypeError, () => builder[subOperationName](input, {axes: outsideAxis}));
+          }
+          for (let axis of notUnsignedLongAxisArray) {
+            assert_false(typeof axis === 'number' && Number.isInteger(axis), `[${subOperationName}]  any of options.axes elements should be of 'unsigned long'`);
```
Comment (fdwr) [inline webnn/resources/utils_validation.js:276]:
```suggestion
            assert_false(typeof axis === 'number' && Number.isInteger(axis), `[${subOperationName}] any of options.axes elements should be of 'unsigned long'`);
```

File: webnn/validation_tests/batch_normalization.https.any.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+// META: script=../resources/utils_validation.js
+// META: timeout=long
+
+'use strict';
+
+promise_test(async t => {
+  for (let dataType of operandDataTypeArray) {
+    const input = builder.input('input', {dataType, dimensions: dimensions2D});
+    const validAxisArray = getAxisArray(dimensions2D);
+    const outsideAxisArray = getOutsideValueArray(unsignedLongType);
```
Comment (fdwr) [inline webnn/validation_tests/batch_normalization.https.any.js:12]: Propose `validAxisArray` and `invalidAxisArray`.

File: webnn/validation_tests/batch_normalization.https.any.js [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+}, "[batchNormalization] DataError is expected if the size of mean.dimensions is not 1");
+
+promise_test(async t => {
+  for (let dataType of operandDataTypeArray) {
+    const input = builder.input('input', {dataType, dimensions: dimensions2D});
+    const validAxisArray = getAxisArray(dimensions2D);
+    for (let axis of validAxisArray) {
+      let size = dimensions2D[axis];
+      const variance = builder.input('variance', {dataType, dimensions: [size]});
+      for (let offset of adjustOffsetArray) {
```
Comment (fdwr) [inline webnn/validation_tests/batch_normalization.https.any.js:77]: Propose `adjustedOffsetsArray`.

File: webnn/validation_tests/batch_normalization.https.any.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      }
+    }
+  }
+}, "[batchNormalization] DataError is expected if mean.dimensions[0] is not equal to input.dimensions[options.axis]");
+
+promise_test(async t => {
+  for (let dataType of operandDataTypeArray) {
+    const input = builder.input('input', {dataType, dimensions: dimensions2D});
+    const validAxisArray = getAxisArray(dimensions2D);
+    for (let axis of validAxisArray) {;
```
Comment (fdwr) [inline webnn/validation_tests/batch_normalization.https.any.js:90]: Extra semicolon `{;`.

**Review by fdwr — COMMENTED** [review]
It's good to have these additional parameter validation tests (and parameter *invalidation* tests too).

**Comment by BruceDai** [comment]
@fdwr I've updated commit to address your comments, and I also rebased code to fix conflicts, please take another look, thanks.

File: webnn/resources/utils_validation.js [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+    operationNameArray = operationName;
+  }
+  for (let subOperationName of operationNameArray) {
+    promise_test(async t => {
+      for (let dataType of operandDataTypeArray) {
+        for (let dimensions of dimensionsArray) {
+          const inputA = builder.input('inputA', {dataType, dimensions});
+          for (let dataTypeB of operandDataTypeArray) {
+            if (dataType !== dataTypeB) {
+              const inputB = builder.input('inputB', {dataType: dataTypeB, dimensions});
```
[the remark this answers is not in the document]
Comment (BruceDai) [reply webnn/resources/utils_validation.js]: Now I use `inputB` by subfixing `inputBIndex` like *inputB1*, *inputB2*, *inputB3* and etc.. for operand name. And also update similar operand name. What's your opinion? Thanks.

File: webnn/resources/utils_validation.js [hunk 6]
```diff
@@ -21,14 +21,16 @@ const dimensions3D = [2, 3, 4];
...
+const adjustOffsetsArray = [
   // Decrease 1
   -1,
   // Increase 1
   1
 ];
 
-const dimensionsArray = [
+// TODO
+// Add more 5+ D dimensions
```
Comment (fdwr) [inline webnn/resources/utils_validation.js:32]:
```suggestion
// Add more 5+ dimensions
```
(no need to repeat dimensions twice)

**Review by fdwr — APPROVED** [review]
👍

**Review by fdwr — APPROVED** [review]

**Review by Honry — APPROVED** [review]
👍

**Comment by BruceDai** [comment]
@Honry I re-rebased code to trigger CIs, let's wait the CIs check status. Thanks.

**Comment by Honry** [comment]
Friendly ping @jgraham, @foolip, anyone can help on the CI failures? Looks like it is irrelevant to this PR.

**Comment by foolip** [comment]
It looks the failures are due to the large number of affected tests. I'll admin merge.

**Comment by BruceDai** [comment]
@foolip Thanks for your great help!

**Pull request merged**